File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] daysArr = new int[progresses.length];
        
        for (int i = 0; i < progresses.length; i++) {
            int left = 100 - progresses[i];
            int days = left / speeds[i];
            
            daysArr[i] = (left % speeds[i] == 0) ? days : days + 1;
        }
        
        List<Integer> answer = new ArrayList<>();
        int prevMaxDays = daysArr[0];
        int count = 0;
        
        for (int i = 0; i < daysArr.length; i++) {
            if (daysArr[i] <= prevMaxDays) {
                count++;
            } else {
                answer.add(count);
                count = 1;
                prevMaxDays = daysArr[i];
            }
        }
        
        answer.add(count);
        
        return answer.stream().mapToInt(i -> i).toArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let answer = [];
    let days = [];
    
    for (let i = 0; i < speeds.length; i++) {
        days.push(Math.ceil((100 - progresses[i]) / speeds[i]));
    }
    
    let max = days[0];
    let cnt = 1;
    
    for (let i = 1; i < days.length; i++) {
        if (days[i] <= max) {
            cnt++;
        } else {
            max = days[i];
            answer.push(cnt);
            cnt = 1;
        }
    }
    
    answer.push(cnt);
    
    return answer;
}